core/src/main/scala/org/http4s/Message.scala

```diff
@@ -55,23 +55,20 @@ sealed trait Message[F[_]] extends MessageOps[F] { self =>
     * @tparam T type of the Body
     * @return a new message with the new body
     */
-  def withBody[T](b: T)(implicit F: Monad[F], w: EntityEncoder[F, T]): F[Self] =
-    w.toEntity(b).flatMap { entity =>
-      val hs = entity.length match {
-        case Some(l) =>
-          `Content-Length`
-            .fromLong(l)
-            .fold(
-              _ =>
-                F.pure {
-                  Message.logger.warn(s"Attempt to provide a negative content length of $l")
-                  w.headers.toList
-              },
-              cl => F.pure(cl :: w.headers.toList))
-        case None => F.pure(w.headers)
-      }
-      hs.map(newHeaders => change(body = entity.body, headers = headers ++ newHeaders))
+  def withBody[T](b: T)(implicit w: EntityEncoder[F, T]): Self = {
```

Comment:
This breaks a lot of things.  I wonder about deprecating this and introducing a new name for the pure one.  How about `withEntity`?

Reply:
I wish I would've had that idea before I changed like 200 lines that depended on it :(

Reply:
jk IJ to the rescue :D